sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetVMsImpl.java

```diff
@@ -86,6 +89,78 @@ public void deleteInstances(Collection<String> instanceIds, boolean forceDeletio
         this.deleteInstancesAsync(instanceIds, forceDeletion).block();
     }
 
+    @Override
+    public void deallocateInstances(Collection<String> instanceIds) {
+        this.deallocateInstancesAsync(instanceIds).block();
+    }
+
+    @Override
+    public Mono<Void> deallocateInstancesAsync(Collection<String> instanceIds) {
+        return this.scaleSet.manager().serviceClient().getVirtualMachineScaleSets().deallocateAsync(
+            this.scaleSet.resourceGroupName(), this.scaleSet.name(),
+            new VirtualMachineScaleSetVMInstanceIDs().withInstanceIds(new ArrayList<>(instanceIds)));
+    }
+
+    @Override
+    public void powerOffInstances(Collection<String> instanceIds, boolean skipShutdown) {
+        this.powerOffInstancesAsync(instanceIds, skipShutdown).block();
+    }
+
+    @Override
+    public Mono<Void> powerOffInstancesAsync(Collection<String> instanceIds, boolean skipShutdown) {
+        return this.scaleSet.manager().serviceClient().getVirtualMachineScaleSets().powerOffAsync(
+            this.scaleSet.resourceGroupName(), this.scaleSet.name(), skipShutdown,
+            new VirtualMachineScaleSetVMInstanceIDs().withInstanceIds(new ArrayList<>(instanceIds)));
+    }
+
+    @Override
+    public void startInstances(Collection<String> instanceIds) {
+        this.startInstancesAsync(instanceIds).block();
+    }
+
+    @Override
+    public Mono<Void> startInstancesAsync(Collection<String> instanceIds) {
+        return this.scaleSet.manager().serviceClient().getVirtualMachineScaleSets().startAsync(
+            this.scaleSet.resourceGroupName(), this.scaleSet.name(),
+            new VirtualMachineScaleSetVMInstanceIDs().withInstanceIds(new ArrayList<>(instanceIds)));
+    }
+
+    @Override
+    public void restartInstances(Collection<String> instanceIds) {
+        this.restartInstancesAsync(instanceIds).block();
+    }
+
+    @Override
+    public Mono<Void> restartInstancesAsync(Collection<String> instanceIds) {
+        return this.scaleSet.manager().serviceClient().getVirtualMachineScaleSets().restartAsync(
+            this.scaleSet.resourceGroupName(), this.scaleSet.name(),
+            new VirtualMachineScaleSetVMInstanceIDs().withInstanceIds(new ArrayList<>(instanceIds)));
+    }
+
+    @Override
+    public void redeployInstances(Collection<String> instanceIds) {
+        this.redeployInstancesAsync(instanceIds).block();
+    }
+
+    @Override
+    public Mono<Void> redeployInstancesAsync(Collection<String> instanceIds) {
+        return this.scaleSet.manager().serviceClient().getVirtualMachineScaleSets().redeployAsync(
+            this.scaleSet.resourceGroupName(), this.scaleSet.name(),
+            new VirtualMachineScaleSetVMInstanceIDs().withInstanceIds(new ArrayList<>(instanceIds)));
+    }
+
+//    @Override
+//    public void reimageInstances(Collection<String> instanceIds) {
```

Comment:
Why we remove reimageInstances() and reimageInstancesAsync()?

Reply:
Reason is that I am not able to get it work in tests.

If I do one instance in array, it is good. But 2, fails.

Same error as https://github.com/hashicorp/terraform-provider-azurerm/issues/14499
So I assume it is server problem.